sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/KeyEncryptionKeyClientBuilder.java

```diff
@@ -0,0 +1,175 @@
+ * {@link TokenCredential credential}).</p>
+ *
+ * <p>The {@link HttpLogDetailLevel log detail level}, multiple custom {@link HttpLoggingPolicy policies} and custom
+ * {@link HttpClient http client} can be optionally configured in the {@link KeyEncryptionKeyClientBuilder}.</p>
+ *
+ * <p>Alternatively, custom {@link HttpPipeline http pipeline} with custom {@link HttpPipelinePolicy} policies
+ * can be specified. It provides finer control over the construction of {@link AsyncKeyEncryptionKey} and {@link
+ * KeyEncryptionKey}</p>
+ *
+ * <p> The minimal configuration options required by {@link KeyEncryptionKeyClientBuilder keyEncryptionKeyClientBuilder}
+ * to build {@link KeyEncryptionKey} are {@link String key identifier}) and
+ * {@link TokenCredential credential}).</p>
+ *
+ * @see KeyEncryptionKeyAsyncClient
+ * @see KeyEncryptionKeyClient
+ */
+@ServiceClientBuilder(serviceClients = {KeyEncryptionKeyClient.class, KeyEncryptionKeyAsyncClient.class})
+public final class KeyEncryptionKeyClientBuilder extends CryptographyClientBuilder implements KeyEncryptionKeyResolver, AsyncKeyEncryptionKeyResolver {
+    private final ClientLogger logger = new ClientLogger(KeyEncryptionKeyClientBuilder.class);
+
+    /**
+     * The constructor with defaults.
+     */
+    public KeyEncryptionKeyClientBuilder() {
+        super();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public KeyEncryptionKey buildKeyEncryptionKey(String keyId) {
+        return new KeyEncryptionKeyClient((KeyEncryptionKeyAsyncClient) buildAsyncKeyEncryptionKey(keyId).block());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<? extends AsyncKeyEncryptionKey> buildAsyncKeyEncryptionKey(String keyId) {
+        this.keyId = keyId;
+        if (Strings.isNullOrEmpty(keyId)) {
+            throw logger.logExceptionAsError(new IllegalStateException(
+                "Json Web Key or jsonWebKey identifier are required to create key encryption key async client"));
+        }
+
+        if (pipeline != null) {
+            return Mono.defer(() -> Mono.just(new KeyEncryptionKeyAsyncClient(keyId, pipeline)));
+        }
+
+        if (credential == null) {
+            throw logger.logExceptionAsError(new IllegalStateException(
+                "Key Vault credentials are required to build the key encryption key async client"));
+        }
+
+        HttpPipeline pipeline = setupPipeline();
+
+        return Mono.defer(() -> Mono.just(new KeyEncryptionKeyAsyncClient(keyId, pipeline)));
+    }
+
+    /**
+     * Sets the credential to use when authenticating HTTP requests.
+     *
+     * @param credential The credential to use for authenticating HTTP requests.
+     * @return the updated {@link KeyEncryptionKeyClientBuilder} object.
+     * @throws NullPointerException if {@code credential} is {@code null}.
+     */
+    public KeyEncryptionKeyClientBuilder credential(TokenCredential credential) {
```

Comment:
As discussed, call super and inheritDoc. Make this class as simple as possible.

Reply:
done

Reply:
fixed